src/lib.rs

```diff
@@ -366,17 +366,20 @@ to write correct code for Unix, at the cost of getting a corner case wrong on
 Windows.
 */
 
-#![cfg_attr(not(feature = "std"), no_std)]
+#![no_std]
```

Comment:
This change requires all items in the prelude to be imported even when linking `std`. It made fixing all of the import errors much easier. I also saw a recommendation in an issue on the Rust issue tracker that preferred this syntax. I think I've seen it used in `wasm-bindgen`.

Reply:
Hmmm, this is not how I've done it in the past. Could you say more about what becomes easier with this?

One thing that this makes kind of a pain is that (AIUI) we can't benefit from std's prelude in tests if we do this, right? That's kind of a bummer and I'd rather have the prelude available in tests.

Reply:
I found the docs that refer to this approach:

https://github.com/rust-lang/cargo/blob/58a961314437258065e23cb6316dfc121d96fb71/src/doc/src/reference/features-examples.md#no_std-support

Reply:
> One thing that this makes kind of a pain is that (AIUI) we can't benefit from std's prelude in tests if we do this, right? That's kind of a bummer and I'd rather have the prelude available in tests.

You can do this too:
```
#![cfg_attr(not(test), no_std)]
#[cfg(feature = "std")]
extern crate std;
```

(or `#[cfg(all(not(test), feature = "std"))]` on the `extern crate std` — I don't think there's a difference)

But given that bstr already has most of the tests in `mod tests` you could just put `use std::prelude::v1::*` at the top of it and get most of the benefit.

It's kind of annoying that there's not a better way to do this — the way the prelude works for no_std is janky, honestly.